packages/app-builder-lib/src/node-module-collector/hoist.ts

```diff
@@ -0,0 +1,1110 @@
+// copy from https://github.com/yarnpkg/berry/blob/master/packages/yarnpkg-nm/sources/hoist.ts
```

Comment:
Just realized we need the license file for copying this. Please copy this to the top of this file: https://github.com/yarnpkg/berry/blob/master/LICENSE.md
>Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.

Reply:
added.